lxd/main_init_interactive.go

```diff
@@ -360,16 +360,27 @@ func (c *cmdInit) askNetworking(config *api.InitPreseed, d lxd.InstanceServer) e
 					continue
 				}
 
-				// Add to the default profile
-				config.Node.Profiles[0].Devices["eth0"] = map[string]string{
-					"type":    "nic",
-					"nictype": "macvlan",
-					"name":    "eth0",
-					"parent":  interfaceName,
-				}
+				// Add to the default profile.
+				network, _, err := d.GetNetwork(interfaceName)
+				if err == nil && network.Managed {
+					// Add managed network.
+					config.Node.Profiles[0].Devices["eth0"] = map[string]string{
+						"name":    "eth0",
+						"type":    "nic",
+						"network": network.Name,
+					}
+				} else {
+					// Add unmanaged network.
+					config.Node.Profiles[0].Devices["eth0"] = map[string]string{
+						"name":    "eth0",
+						"type":    "nic",
+						"nictype": "macvlan",
+						"parent":  interfaceName,
+					}
 
-				if shared.PathExists(fmt.Sprintf("/sys/class/net/%s/bridge", interfaceName)) {
-					config.Node.Profiles[0].Devices["eth0"]["nictype"] = "bridged"
+					if shared.PathExists(fmt.Sprintf("/sys/class/net/%s/bridge", interfaceName)) {
```

Comment:
@MusicDin could we rework this section so it just pulls a list of networks from LXD client and then looks at their network type to decide what to do rather than probing /sys/class here (its what LXD is doing anyway when pulling a list of networks).

Reply:
Yes. Thanks for suggestion